### DISCUSSION TITLE
Understanding of the Instance segmentation

### DISCUSSION THREAD

**Discussion by kadattack** [bowenc0221/panoptic-deeplab#88] [Q&A] [answered]
Hello, 
I am trying to understand the full working of the Panoptic-DeepLab architecture, however I am really struggling at understanding how Instance segmentation works. Particularly how center regression and instance center work. I'm trying to wrap my head around how center heatmaps are predicted and how ground truth 2-D Gaussian encoded heatmaps are generated during training and how that is used to train the model.
Can you please tell me where that part of the code is implemented, so that I may perhaps understand how it works through code?

**Comment by bowenc0221** [comment, marked as answer]
Training target generation code: https://github.com/facebookresearch/detectron2/blob/master/projects/Panoptic-DeepLab/panoptic_deeplab/target_generator.py

Model definition: https://github.com/facebookresearch/detectron2/blob/master/projects/Panoptic-DeepLab/panoptic_deeplab/panoptic_seg.py

Post-processing: https://github.com/facebookresearch/detectron2/blob/master/projects/Panoptic-DeepLab/panoptic_deeplab/post_processing.py

**Comment by kadattack** [comment]
I would like to ask one more question regarding implementation of the Backbone. The paper https://arxiv.org/pdf/1911.10194.pdf describes different variants of the HRNet implementation (HRNet, HRNet+, HRNet-Wider+). 
Which variant was implemented in the code if you select "d2_hrnet" as the MODEL.WEIGHTS.BACKBONE.NAME variable like in the (panoptic_deeplab_H_48_os16_mg124_poly_90k_bs32_crop_512_1024.yaml) config?
If I take a look at panoptic-deeplab-master\tools_d2\d2\backbone.py file, it doesn't look like it's using the (HRNet-Wider+) version.
